### PR TITLE
Free text message and send

### PR DESCRIPTION
**Pull request by elayabharath**
+ clear icon

### CODE REVIEW

File: lib/components/messages/freetext_message_send.dart [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+      ..className = "freetext-send--container"
+      ..append(_sendButton);
+
+    renderElement = DivElement()
+      ..className = "freetext-message-send--container"
+      ..append(textareaWrapper)
+      ..append(sendButtonWrapper);
+  }
+
+  void _enableDisableButtons() {
```
Comment (marianamarasoiu) [inline lib/components/messages/freetext_message_send.dart:61]: nit: rename this to `_enableOrDisableButtons`

File: lib/components/messages/freetext_message_send.dart [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+  FreetextMessageSendView(this._text) {
+    _onSendController = StreamController();
+    _onSend = _onSendController.stream;
+
+    _textArea = TextAreaElement()
+      ..defaultValue = _text
+      ..placeholder = "Type your message..."
+      ..onInput.listen((e) {
+        _text = (e.target as TextAreaElement).value;
```
Comment (marianamarasoiu) [inline lib/components/messages/freetext_message_send.dart:27]: nit:
```suggestion
        _text = _textArea.value;
```

**Review by marianamarasoiu — APPROVED** [review]
thanks EB, looks good, just a few nits below and a suggestion to rename the CSS classes to read like below:

```
<div class="message-editor-with-send">
  <div class="message-editor-with-send__message">
    <textarea placeholder="Type your message...">Default text</textarea>
    <button class="clear-button"><img src="assets/icons/clear.svg"></button>
  </div>
  <div class="message-editor-with-send__send-button">
    <button>Send</button>
  </div>
</div>
```

I think the above follows a bit more closely the BEM conventions for "__" to refer to elements while "--" refers to modifiers (not sure how that worked for containers?), and it also makes the hierarchy a bit clearer. What do you think?

**Comment by elayabharath** [comment]
thanks @marianamarasoiu addressed all the suggestions

**Pull request merged**